### ISSUE TITLE
superfluous-parens / C0325 miss alarm

### ISSUE THREAD

**Issue by bebound** [pylint-dev/pylint#10084]
### Bug description

```python
This code should raise C0325:

a = 1

if (1 == 1 and 2 == 1):
    pass

if (a is None and a is not None):
    pass
```

### Configuration

```ini

```

### Command used

```shell
pylint --disable=all --enable=C0325 kk.py
```

### Pylint output

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

Raise C0325

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.5
Python 3.12.7 (tags/v3.12.7:0b05ead, Oct  1 2024, 03:06:41) [MSC v.1941 64 bit (AMD64)]
```

### Additional dependencies

```python

```

**Comment by ChandanChainani** [comment]
Hi @bebound

Below attached code which is part of `pylint` as per which above statements are correct.

https://github.com/pylint-dev/pylint/blob/abb81878fd114f309297b35e1d392bee0d196fbf/tests/functional/u/unnecessary/unnecessary_negation.py#L44

cc: @Pierre-Sassoulas

**Comment by Pierre-Sassoulas** [comment]
Thank you for the investigation. The git blame could explain what the reasoning was at the time and if help with the decision going forward. (Can't check right now I'm on mobile)

**Comment by ranadheerg** [comment]
I would like to work on this issue if it is open

**Comment by skdas20** [comment]
@ranadheerg are you still working on this? No worries if life got in the way — just checking since it's been a few weeks. If it's free, I'd be happy to pick it up: #11081 (the `in` sibling case) looks like a good template to extend for the `and`/`or` case, and I have a working patch locally. @Pierre-Sassoulas happy to defer to whoever is actively on it.

**Comment by ranadheerg** [comment]
@skdas20 yes please! Take it and run. I can't give this the time it needs right now, and if you've got a patch working, that's way better than it sitting idle. Thanks for picking it up 🙏

**Comment by skdas20** [comment]
Thanks @ranadheerg, appreciate you saying so rather than leaving it ambiguous — that's the blocker cleared from my side.

Patch is already up as #11190: it narrows the blanket `and`/`or` exemption in `_check_keyword_parentheses` so parentheses wrapping an entire condition are reported, while keeping them accepted where they're actually required — after `not`, and where the parenthesised expression is an operand of a following binary operator such as `in`. Functional tests cover both directions, and running the updated checker over pylint's own source produces no new messages beyond the three that already exist on `main`.
